src/plugins/discover/public/application/angular/redirect.ts

```diff
@@ -16,8 +16,21 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+import { getAngularModule, getServices, getUrlTracker } from '../../kibana_services';
 
-export {
-  ProcessedImportResponse,
-  processImportResponse,
-} from '../../../../plugins/saved_objects_management/public/lib'; // eslint-disable-line @kbn/eslint/no-restricted-paths
+getAngularModule().config(($routeProvider: any) => {
+  $routeProvider.otherwise({
+    redirectTo: () => {
+      const path = window.location.hash.substr(1);
+      getUrlTracker().restorePreviousUrl();
+      setTimeout(() => {
+        const { kibanaLegacy } = getServices();
+        const { navigated } = kibanaLegacy.navigateToLegacyKibanaUrl(path);
+        if (!navigated) {
+          kibanaLegacy.navigateToDefaultApp();
+        }
+      }, 0);
```

Comment:
Can you please describe exactly what this and the other `setTimeout(..., 0)` calls are waiting for with comments in the code? Is there a reason we can't establish a more direct relationship with whatever action this code is depending on? (wait for a promise, event, callback, something?)

Reply:
Good point, it's waiting for angular because doing it sync messed up the internal state handling of ngRoute. I replaced it with an `$applyAsync`.